icgcget/clients/portal_client.py

```diff
@@ -63,6 +65,37 @@ def __init__(self, verify):
         self.logger = logging.getLogger('__log__')
         self.verify = verify
 
+
+    """
+    Extract the ordered repository list from the manifest_id
+    """
+    def __get_repos_from_manifest_id(self, manifest_id, api_url):
+        pre_request = (api_url + 'manifests/' + manifest_id + '?format=json')
+        try:
+            entity_set = call_api(pre_request, verify=self.verify)
+            return entity_set["repos"]
+        except ApiError as ex:
+            self.logger.error('There was an issue extracting the repo priority from manifestId "{}"'.format(manifest_id) +
+                              ' Please check your manifest id')
+            raise ApiError(ex.request_string, ex.message, ex.code)
+
+
+    # Verifies the repo list in the manifest id matches the manifest id
+    def check_repo_list(self, manifest_id, api_url, configured_repo_list):
+        # Extract repo list from manifest_id
+        manifest_repo_list = self.__get_repos_from_manifest_id(manifest_id, api_url)
+
+        # Find the repositories that were not configured
+        missing_list = []
+        for manifest_repo in manifest_repo_list:
+            if manifest_repo not in configured_repo_list:
+                missing_list.append(manifest_repo)
+
+        if len(missing_list) > 0:
+            message = 'The following repositories from the manifest id "{}" are not configured: {}'.format(manifest_id,set(missing_list))
+            raise ApiError(api_url, message, 400)
```

Comment:
This shouldn't be an API error as the API at this point should have already returned successfully. 

I think the `click` package has some useful exceptions. Try maybe `click.UsageError`

A good rule of thumb is, if you cannot retrieve the status code for your exception context in the client and have to set the status code yourself, you probably shouldn't be using something that reports HTTP error codes.